src/SLCore.IntegrationTests/FileAnalysisTests.cs

```diff
@@ -109,25 +110,77 @@ public async Task StandaloneRuleConfig_CtorParamsAboveThreshold_JavaScriptActive
         issuesByFileUri[new FileUri(GetFullPath(OneIssueRuleWithParamPath))].Should().HaveCount(1);
     }
 
+    [TestMethod]
+    public async Task StandaloneRuleConfig_JsLetRuleIsDisableInSettingsFile_JavaScriptAnalysisShouldIgnoreIssueOnInitialization()
+    {
+        var letRuleConfig = CreateInactiveRuleConfig(LetRuleId);
+
+        var issuesByFileUri = await RunFileAnalysis(TwoJsIssuesPath, initialRulesConfig: letRuleConfig, updatedRulesConfig:null);
+
+        issuesByFileUri.Should().HaveCount(1);
+        issuesByFileUri[new FileUri(GetFullPath(TwoJsIssuesPath))].Should().HaveCount(1);
+    }
+
+    [TestMethod]
+    public async Task StandaloneRuleConfig_CloudSecretsRuleIsDisabledInSettingsFile_SecretsAnalysisShouldIgnoreIssueOnInitialization()
+    {
+        var secretsRuleConfig = CreateInactiveRuleConfig(CloudSecretsRuleId);
+
+        var issuesByFileUri = await RunFileAnalysis(ThreeSecretsIssuesPath, initialRulesConfig:secretsRuleConfig, updatedRulesConfig:null);
+
+        issuesByFileUri.Should().HaveCount(1);
+        issuesByFileUri[new FileUri(GetFullPath(ThreeSecretsIssuesPath))].Should().HaveCount(1);
+    }
+
     private async Task<Dictionary<FileUri, List<RaisedIssueDto>>> RunFileAnalysis(string fileToAnalyzeRelativePath, Dictionary<string, StandaloneRuleConfigDto> ruleConfigByKey = null)
     {
-        const string configScopeId = "ConfigScope1";
+        return await RunFileAnalysis(fileToAnalyzeRelativePath, null, ruleConfigByKey);
```

Comment:
why overload and not optional parameters?

Reply:
Because they are more speaking. Optional parameters some disadvantages, like the fact  the number of optional parameters tend to increase (I have seen that a lot of unit tests methods have > 7 optional parameter), which is makes code very ugly and difficult to maintain.

Reply:
But they're not more speaking if you use the same name for them. Rename the first one to `RunFileAnalysisWithDefaultRuleConfiguration`.

Reply:
Done, but the renaming is a little bit different.